File: Graph/ScriptTemplate/Random/CGRandom.lua

local CGRandom = CGRandom or {}
CGRandom.__index = CGRandom

function CGRandom.new()
    local self = setmetatable({}, CGRandom)
    self.inputs = {}
    self.outputs = {}
    self.nexts = {}
    return self
end

function CGRandom:setNext(index, func)
    self.nexts[index] = func
end

function CGRandom:setInput(index, func)
    self.inputs[index] = func
end

function CGRandom:getOutput(index)
    return self.outputs[index]
end

function CGRandom:execute(index)    
    self.outputs[1] = math.random(self.inputs[1](), self.inputs[2]())
    if self.nexts[0] ~= nil then
        self.nexts[0]()
    end
end

return CGRandom

File: Graph/ScriptTemplate/Random/CGRandom.js
/**
 * @file CGRandom.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGRandom.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGRandom extends BaseNode {
  constructor() {
    super();
  }

  getOutput(index) {
    if (!this.inputs[0] === undefined || !this.inputs[1] === undefined) {
      return 0;
    }
    if (this.inputs[0]() === undefined || !this.inputs[1]() === undefined) {
      return 0;
    }
    const upper = Math.max(this.inputs[0](), this.inputs[1]());
    const lower = Math.min(this.inputs[0](), this.inputs[1]());
    return Number((Math.random() * Math.abs(upper - lower) + lower).toFixed(3));
  }
}

exports.CGRandom = CGRandom;
